ess-dmsc/nexus-constructor
# Making pixel models inherit from QObject rather than QAbstractListModel

### Issue

Relates to #153 

### Description of work

Refactored 2 of the pixel models that are wrapping pixel mapping types to inherit from `QObject` and use `Properties` 

- I tried using lambdas for the getters/setters, but there's no easy way as `self` is out of scope when declaring the `Property`.
- `PixelMappingModel` still uses a `QAbstractListModel`, however I think this is the appropriate thing to use in this scenario, at least for now until we figure a bit more about how to use normal `QObject`s with stuff like `ListView`s 
 - `change_value` is no longer used in this file. hooray! 

Have added some tests to cover the new changes, note that not all of the getters/setters are tested nor do they really need to be, as they don't cover much logic. 

### Acceptance Criteria 
To test these changes: 
`SinglePixelModel`:
- Add a component, choose `monitor` type and enter a pixel id that is not 0
- Check in the JSON output that this has been reflected.

`PixelGridModel`
- Add a component, choose `detector` and `Repeatable grid`. 
- Use the `cube.off` file in `tests/` for the geometry file
- change the pixel grid to use something like 2 columns and 3 rows, check that the shape is a rectangle that is 2 wide and 3 high.
- repeat but test with row heights and widths 
- repeat and check that pixel ids are reflected in the JSON 

`PixelMappingModel` has not changed in any way so no need to test this unless you really want to. 


### Nominate for Group Code Review

- [ ] Nominate for code review
